server/src/main/java/org/elasticsearch/indices/IndexOpenException.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.indices;
+
+import org.elasticsearch.ElasticsearchException;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.index.Index;
+import org.elasticsearch.rest.RestStatus;
+
+import java.io.IOException;
+
+/**
+ * Exception indicating that index can not be opened
+ */
+public class IndexOpenException extends ElasticsearchException {
```

Comment:
I do not think we need a new exception type? Is there an existing one that we can reuse?

Reply:
@jasontedor Thanks Jason for the review! I could not find any relevant exception. `ILLEGAL_INDEX_SHARD_STATE_EXCEPTION` - 404, `INDEX_CLOSED_EXCEPTION` - 400? `ElasticsearchException` that was used in the code before produces 500, while I think we want to produce 400, as it is a user's error to open an index with broken settings. What would you suggest?

Reply:
Why not `IllegalArgumentException`?